src/Features/Core/Portable/AddConstructorParametersFromMembers/AddConstructorParametersFromMembersCodeRefactoringProvider.State.cs

```diff
@@ -87,18 +87,22 @@ private class State
             {
                 var parameterNames = parameters.SelectAsArray(p => p.Name);
                 return containingType.InstanceConstructors
-                    .Where(constructor => AreParametersContainedInConstructor(constructor, parameterNames))
+                    .Where(constructor => IsApplicableConstructor(constructor, parameterNames))
                     .OrderByDescending(constructor => constructor.Parameters.Length)
                     .FirstOrDefault();
             }
 
-            private bool AreParametersContainedInConstructor(
+            private bool IsApplicableConstructor(
                 IMethodSymbol constructor,
                 ImmutableArray<string> parametersName)
             {
                 var constructorParams = constructor.Parameters;
-                return constructorParams.Length > 0
-                    && constructorParams.All(parameter => parameter.RefKind == RefKind.None)
+                if (constructorParams.Length <= 0)
+                {
+                    return !constructor.IsImplicitlyDeclared;
```

Comment:
What's the behavior if we only have an implicit default ctor? Refactoring will not be available?

Reply:
yeah, i would prefer it's avialable.

Reply:
See this: https://github.com/dotnet/roslyn/pull/33624#discussion_r260978038